src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/ReferenceDataResourceTest.java

```diff
@@ -181,4 +185,74 @@ public void testReadDomainReverseLookupWithWildcard() {
         assertThatJsonFileAndStatus(response, 200, "multiple_ref_data_reverse_lookup.json");
     }
 
+    @Nested
+    @DisplayName("GET /domains")
+    class GetDomainsTest {
+        @Test
+        @DisplayName("must have a valid token to access endpoint")
+        void mustHaveAValidTokenToAccessEndpoint() {
+            assertThat(getDomains(null).getStatusCode()).isEqualTo(HttpStatus.UNAUTHORIZED);
+        }
+
+        @Test
+        @DisplayName("can have any role to access endpoint")
+        void canHaveAnyRoleToAccessEndpoint() {
+            final var token = authTokenHelper.someClientUser("ROLE_BANANAS");
+
+            assertThat(getDomains(token).getStatusCode()).isEqualTo(HttpStatus.OK);
+        }
+
+        @Test
+        @DisplayName("will return a list of domains")
+        void willReturnAListOfDomains() {
+            final var token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
+
+            final var response = getBodyAsJsonContent(getDomains(token));
+
+            assertThat(response).hasJsonPathArrayValue("$");
+            assertThat(response).extractingJsonPathNumberValue("$.length()").isEqualTo(430);
```

Comment:
dubious about this one - looks like the test will be fragile as any new domain will cause it to fail.  Could you do a `> 300` and `< 600` instead?